solr/core/src/java/org/apache/solr/response/transform/DeeplyNestedChildDocTransformer.java

```diff
@@ -190,8 +179,18 @@ private String getLastPath(String path) {
     return path.substring(path.lastIndexOf(PATH_SEP_CHAR.charAt(0)) + 1);
   }
 
-  private String trimSuffixFromPaths(String path) {
-    return path.replaceAll("#\\d|#", "");
+  private String trimIfSingleDoc(String path) {
```

Comment:
I'll assume this is still WIP as we discussed wanting to use a Set<String> of child docs, and thus we wouldn't trim conditionally for single doc.

Reply:
Do we really need a set though?
If we trim only the array docs we can then easily figure out at insert time whether the item is a single doc or an array, optimizing the insert time, and removing the need for a set

Reply:
Oh gotcha -- sure.  It does complicate an explanation of what that String is, as it won't simply be a label, but that's okay.  Please add comments where we declare the field to explain this stuff. (e.g.. what the outer String is (with example), what the intermediate string is (with 2 examples)).